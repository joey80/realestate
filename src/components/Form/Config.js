export const API = {
    key: 'Qzv7AxmxjkkqtzYKcRAEsaIxXlOudY'
};