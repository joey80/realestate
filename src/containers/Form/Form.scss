@import '../../styles/main';

.form {
    padding: 1rem;
    width: 100%;

    &__button {
        align-items: center;
        background-color: #868e96;
        border-radius: 5px;
        color: #fff;
        display: flex;
        font-weight: 800;
        padding: 0.5rem 1.1rem;
        transition: all 0.3s ease-in-out;

        &:hover {
            background-color: #362469;
            transition: all 0.3s ease-in-out;
        }
    }

    &__container {
        align-items: flex-end;
        background-color: #f4f4f4;
        border-radius: 12px;
        display: flex;
        flex-direction: column;
        margin: 2rem auto !important;
        opacity: 0.9;
        position: relative;
        width: 50%;
    }

    & h1 {
        color: #444;
        font-family: 'Montserrat', Helvetica, Arial, sans-serif;
        font-size: 1.25rem;
        font-weight: 400;
        letter-spacing: -0.05rem;
        line-height: 1.7;
    }

    & label {
        background-color: #f4f4f4;
        color:rgb(68, 68, 68);
        display: inline-block;
        font-size: 0.8rem;
        font-weight: 800;
        padding: 0rem 1.1rem;
        transform: translate(20px, 33px);
        z-index: 9;
    }

    &__section {
        border: 1px solid rgba(172, 170, 170, 0.2);
        border-top: 4px solid #362469;
        border-radius: 12px;
        box-shadow: 0 0 20px #a5a5a52f;
        margin: 2rem auto;
        padding: 2rem;
        width: 90%;

        &:last-of-type {
            align-items: center;
            border: 0;
            box-shadow: none;
            display: flex;
            justify-content: space-between;
            margin: 0 auto;
            padding: 0 2rem 2rem 2rem;
        }
    }
}