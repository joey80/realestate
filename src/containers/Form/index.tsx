import React from 'react';
import { batch, useDispatch, useSelector } from 'react-redux';
import axios from 'axios';
import { INPUT, MODAL, RESULTS } from 'src/actions/constants';
import InputContainer from 'src/containers/InputContainer';
import ModalContainer from 'src/containers/ModalContainer';
import SelectContainer from 'src/containers/SelectContainer';
import { hasValidData } from 'src/utils/Helper';
import { RootState } from 'src/reducers';
import './styles.scss';

const Form = () => {
  const dispatch = useDispatch();
  const searchLocation = useSelector((state: RootState) => state.Input.searchLocation);

  const handleFormSubmit = async (event: React.FormEvent) => {
    event.preventDefault();

    if (hasValidData(searchLocation)) {
      dispatch({ type: MODAL.START_LOADING });
      handleZillow();
    } else {
      window.alert('All Fields Are Required');
    }
  };

  const handleZillow = async () => {
    const zillowResults = await axios('/.netlify/functions/zillow', {
      method: 'POST',
      data: searchLocation,
    });

    if (zillowResults) {
      batch(() => {
        dispatch({ type: RESULTS.SAVE_RESULTS, newResults: zillowResults.data.body });
        dispatch({ type: MODAL.STOP_LOADING });
        dispatch({ type: INPUT.CLEAR_SEARCH_VALUES });
      });
    }
  };

  return (
    <div className='form'>
      <ModalContainer />
      <form className='form__container' onSubmit={handleFormSubmit}>
        <div className='form__section'>
          <h1>Where Would You Like To Search For A Property?</h1>
          <InputContainer
            errorMessage='Please enter a street address'
            label='streetAddress'
            name='Street Address'
          />
          <InputContainer label='city' name='City' errorMessage='Please enter a city name' />
          <SelectContainer
            defaultValue='Select A State'
            errorMessage='Please select a state'
            label='state'
            value={searchLocation.state}
          />
          <InputContainer
            errorMessage='Please enter a valid 5 digit zip code'
            label='zipCode'
            maxLength={5}
            name='Zip Code'
          />
        </div>
        <div className='form__section'>
          <div>* All fields are required</div>
          <button type='submit' className='form__button'>
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default Form;
