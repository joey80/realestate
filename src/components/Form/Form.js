import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import fetch from 'isomorphic-fetch'
import Input from '../Input/Input';
import Modal from '../Modal/Modal';
import Select from '../Select/Select';
import { checkIfNull } from './Helper';
import { API } from '../../env/API';
import './Form.scss';

const convert = require('xml-js');

const mapStateToProps = state => ({
  searchLocation: state.Input.searchLocation,
  results: state.Results.searchResults
});

const mapDispatchToProps = dispatch => ({
  startLoading: () => dispatch({ type: 'START_LOADING' }),
  stopLoading: () => dispatch({ type: 'STOP_LOADING' }),
  saveResults: value => dispatch({ type: 'SAVE_RESULTS', newResults: value }),
  clearSearchValues: () => dispatch({ type: 'CLEAR_SEARCH_VALUES' })
});

export const Form = props => {

  const handleClick = e => {
    e.preventDefault();

    const callAPI = () => {
      const zswid = API.zillow.zwsid;
      const proxy = 'https://cors-anywhere.herokuapp.com/';
    
      fetch(`${proxy}http://www.zillow.com/webservice/GetDeepSearchResults.htm?zws-id=${zswid}&address=${props.searchLocation.streetAddress}&citystatezip=${props.searchLocation.city}+${props.searchLocation.state}+${props.searchLocation.zipCode}`)
      .then(res => res.text())
      .then(res => {
        const rawXML = convert.xml2json(res, { compact: true, spaces: 2 });
        return JSON.parse(rawXML);
      })
      .then(res => props.saveResults(res['SearchResults:searchresults'].response.results.result))
      .then(() => {
        props.stopLoading();
        props.clearSearchValues();
      });
    };

    if (checkIfNull(props.searchLocation) === true) {
      window.alert('All Fields Are Required');
    } else {
      props.startLoading();
      callAPI();
    }
  };

  return (
    <div className="form">
      <Modal />
      <form className="form__container">
        <div className="form__section">
          <h1><strong>Hello!</strong><br />
          Where Would You Like To Search For A Property?</h1>
          <Input
            label="streetAddress"
            name="Street Address"
            type="text"
            errorMessage="Please enter a street address"
          />
          <Input
            label="city"
            name="City"
            type="text"
            errorMessage="Please enter a city name"
          />
          <Select />
          <Input
            label="zipCode"
            name="Zip Code"
            type="text"
            errorMessage="Please enter a valid 5 digit zip code"
          />
        </div>
        <div className="form__section">
          <div>* All fields are required</div>
          <button
            type="submit"
            onClick={ e => { handleClick(e) }}
            className="form__button">
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

Form.propTypes = {
  searchLocation: PropTypes.object,
  startLoading: PropTypes.func,
  saveResults: PropTypes.func,
  stopLoading: PropTypes.func,
  clearSearchValues: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Form);