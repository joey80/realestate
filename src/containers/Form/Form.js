import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { INPUT, MODAL, RESULTS } from '../../actions/constants';
import { zillowAPI } from '../../utils/zillowAPI';
import InputContainer from '../InputContainer/InputContainer';
import ModalContainer from '../ModalContainer/ModalContainer';
import SelectContainer from '../SelectContainer/SelectContainer';
import { checkIfNull } from '../../utils/Helper';
import './Form.scss';

const Form = () => {
  const dispatch = useDispatch();
  const searchLocation = useSelector(state => state.Input.searchLocation);

  const handleClick = async e => {
    e.preventDefault();

    if (checkIfNull(searchLocation) === true) {
      window.alert('All Fields Are Required');
    } else {
      dispatch({ type: MODAL.START_LOADING });
      const zillow = await zillowAPI(searchLocation);

      if (zillow) {
        dispatch({ type: RESULTS.SAVE_RESULTS, newResults: zillow });
        dispatch({ type: MODAL.STOP_LOADING });
        dispatch({ type: INPUT.CLEAR_SEARCH_VALUES });
      }
    }
  };

  return (
    <div className='form'>
      <ModalContainer />
      <form className='form__container'>
        <div className='form__section'>
          <h1>
            <strong>Hello!</strong>
            <br />
            Where Would You Like To Search For A Property?
          </h1>
          <InputContainer
            label='streetAddress'
            name='Street Address'
            type='text'
            errorMessage='Please enter a street address'
          />
          <InputContainer
            label='city'
            name='City'
            type='text'
            errorMessage='Please enter a city name'
          />
          <SelectContainer
            defaultOption='Select A State'
            value={searchLocation.state}
            label='state'
            errorMessage='Please select a state'
          />
          <InputContainer
            label='zipCode'
            name='Zip Code'
            type='text'
            errorMessage='Please enter a valid 5 digit zip code'
          />
        </div>
        <div className='form__section'>
          <div>* All fields are required</div>
          <button
            type='submit'
            onClick={e => {
              handleClick(e);
            }}
            className='form__button'
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default Form;
