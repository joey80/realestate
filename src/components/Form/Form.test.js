import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Form } from './Form';

configure({ adapter: new Adapter() });

describe('Main form', () => {
  it('should render correctly', () => {
    const component = shallow(<Form />);
  
    expect(component).toMatchSnapshot();
  });
});