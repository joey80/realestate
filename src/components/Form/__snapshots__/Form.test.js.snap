// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main form should render correctly 1`] = `
<div
  className="form"
>
  <Connect(Modal) />
  <form
    className="form__container"
  >
    <div
      className="form__section"
    >
      <h1>
        <strong>
          Hello!
        </strong>
        <br />
        Where Would You Like To Search For A Property?
      </h1>
      <Connect(Input)
        errorMessage="Please enter a street address"
        label="streetAddress"
        name="Street Address"
        type="text"
      />
      <Connect(Input)
        errorMessage="Please enter a city name"
        label="city"
        name="City"
        type="text"
      />
      <Connect(Select) />
      <Connect(Input)
        errorMessage="Please enter a valid 5 digit zip code"
        label="zipCode"
        name="Zip Code"
        type="text"
      />
    </div>
    <div
      className="form__section"
    >
      <div>
        * All fields are required
      </div>
      <button
        className="form__button"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
